src/stores/BoxLinksStore.ts

```diff
@@ -0,0 +1,133 @@
+/** *****************************************************************************
+ * Copyright 2009-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { chain, Dictionary, keyBy, mapValues, sortBy, uniqBy } from "lodash";
+import { action, computed, makeObservable, observable } from "mobx";
+import { IBoxConnections, IPinConnections } from "../components/links/BoxConnections";
+import { convertToExtendedLink } from "../helpers/link";
+import { BoxEntity, ExtendedConnectionOwner, isBoxEntity } from "../models/Box";
+import FileBase from "../models/FileBase";
+import { isBoxLinksDefinition, Link, LinksDefinition } from "../models/LinksDefinition";
+import { BoxesStore } from "./BoxesStore";
+import { RequestsStore } from "./RequestsStore";
+import { SelectedBoxStore } from "./SelectedBoxStore";
+
+export class BoxLinksStore {
+
+	linkBoxes: LinksDefinition[] = [];
+
+	public get links(): Link<ExtendedConnectionOwner>[] {
+		if (this.linkBoxes === null) return [];
+
+		return this.linkBoxes.flatMap(linkBox => {
+			if (linkBox.spec['boxes-relation']) {
+				return [
+					...(linkBox.spec['boxes-relation']['router-mq']
+						? linkBox.spec['boxes-relation']['router-mq'].map(link =>
+								convertToExtendedLink(link, 'mq'),
+						  )
+						: []),
+					...(linkBox.spec['boxes-relation']['router-grpc']
+						? linkBox.spec['boxes-relation']['router-grpc'].map(link =>
+								convertToExtendedLink(link, 'grpc'),
+						  )
+						: []),
+				];
+			}
+			return [];
+		});
+	}
+
+	public get separatedLinks(): Array<IBoxConnections | null> {
```

Comment:
That's not separatedLinks.
That should be selectedBoxConnections;
And it's return type is ` [IBoxConnections, IBoxConnections]  | [null, null]`